node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Preparer.java

```diff
@@ -54,34 +53,19 @@ public List<Node> prepare(ApplicationId application, ClusterSpec cluster, NodeSp
      // active config model which is changed on activate
     private List<Node> prepareNodes(ApplicationId application, ClusterSpec cluster, NodeSpec requested) {
         LockedNodeList allNodes = groupPreparer.createUnlockedNodeList();
-        List<Node> surplusNodes = findNodesInRemovableGroups(application, requested.groups(), allNodes);
-        List<Node> accepted = groupPreparer.prepare(application, cluster, requested, surplusNodes, allNodes);
+        List<Node> accepted = groupPreparer.prepare(application, cluster, requested, allNodes);
         if (requested.rejectNonActiveParent()) { // TODO: Move to NodeAllocation
             NodeList activeHosts = allNodes.state(Node.State.active).parents().nodeType(requested.type().hostType());
             accepted = accepted.stream()
                                .filter(node -> node.parentHostname().isEmpty() || activeHosts.parentOf(node).isPresent())
                                .toList();
         }
-        return accepted.stream().filter(node -> ! surplusNodes.contains(node)).collect(Collectors.toList());
+        return new ArrayList<>(accepted);
```

Comment:
Ok, but in general I think it's cleaner to pass around immutable collections and create mutable ones in the same place it's mutated.